packages/boot/src/__tests__/acceptance/crud-rest-builder.acceptance.ts

```diff
@@ -0,0 +1,168 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {ApplicationConfig} from '@loopback/core';
+import {juggler, RepositoryMixin} from '@loopback/repository';
+import {RestApplication} from '@loopback/rest';
+import {
+  CrudRestComponent,
+  defineCrudRepositoryClass,
+} from '@loopback/rest-crud';
+import {expect, givenHttpServerConfig, TestSandbox} from '@loopback/testlab';
+import {resolve} from 'path';
+import {BootMixin, ModelApiBooter} from '../..';
+import {Product} from '../fixtures/product.model';
+
+describe('CRUD rest builder acceptance tests', () => {
+  let app: BooterApp;
+  const SANDBOX_PATH = resolve(__dirname, '../../.sandbox');
+  const sandbox = new TestSandbox(SANDBOX_PATH);
+
+  beforeEach('reset sandbox', () => sandbox.reset());
+  beforeEach(givenAppWithDataSource);
+
+  afterEach(stopApp);
+
+  it('binds the controller and repository to the application', async () => {
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/product.model.js'),
+      'models/product.model.js',
+    );
+
+    await sandbox.writeTextFile(
+      'model-endpoints/product.rest-config.js',
+      `
+const {Product} = require('../models/product.model');
+module.exports = {
+  model: Product,
+  pattern: 'CrudRest',
+  dataSource: 'db',
+  basePath: '/products',
+};
+      `,
+    );
+
+    // Boot & start the application
+    await app.boot();
+    await app.start();
+
+    expect(app.getBinding('repositories.ProductRepository').key).to.eql(
+      'repositories.ProductRepository',
+    );
+
+    expect(app.getBinding('controllers.ProductController').key).to.eql(
+      'controllers.ProductController',
+    );
+  });
+
+  it('uses bound repository class if it exists', async () => {
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/product.model.js'),
+      'models/product.model.js',
+    );
+
+    await sandbox.writeTextFile(
+      'model-endpoints/product.rest-config.js',
+      `
+const {Product} = require('../models/product.model');
+module.exports = {
+  model: Product,
+  pattern: 'CrudRest',
+  dataSource: 'db',
+  basePath: '/products',
+};
+      `,
+    );
+
+    const ProductRepository = defineCrudRepositoryClass(Product);
```

Comment:
If the repository will be created and bound when the config exists, what's the point of creating it here again?
Or do we want to test that if a custom product repository is created and bound to the app, the default one won't be created? Then I think the test needs to make the ProductRepository something different than the default one.

Reply:
> Or do we want to test that if a custom product repository is created and bound to the app, the default one won't be created? Then I think the test needs to make the ProductRepository something different than the default one.

This was the intention. I also tried creating a separate repository class and using that one but thinking about it now, I don't think there's a way to differentiate between them since their bindings are the same, so I might remove this test.

Reply:
>  I also tried creating a separate repository class and using that one but thinking about it now, I don't think there's a way to differentiate between them since their bindings are the same, so I might remove this test.

IMO, we should add the test back and modify the assertions to verify that the repository binding is using the user-provided repository class, not a repository class created by the model-api-builder. You can use `valueConstructor` property to access the class passed to `.toClass()` method when the `Binding` was configured.